src/transport/url.js

```diff
@@ -35,9 +35,22 @@ const send = ({ uriHandler, messageToURI = defaultMessageToURI } = {}) => {
  *  @return   {Object}   A response object if repsonse is available, otherwise null.
  */
 const getResponse = () => {
-  const res = window.location.hash.slice(1)
-  window.location.hash = ''
-  return parseResponse(res)
+  const hash = window.location.hash.startsWith('?') ? window.location.hash.slice(1) : window.location.hash
+  const response = parseResponse(hash)
+  window.location.hash = removeUportHashParams(hash)
```

Comment:
This will trigger a new URL event. We should only change this if we actually have something thats related to uport

Reply:
fixed

Reply:
Fixed this behavior with commit `16e7827`.

Do you have any suggestions for how to test this?  The problem is that the event gets triggered by assigning `window.location.hash = 'new hash'` when run in a browser, but creating a spy on `global.window.onhashchange` does not get called when run in a test.  If I simply check the value of the hash before and after, it won't be able to tell if an assignment was made that triggers the event in a browser but does not change the value.

Reply:
Example of a test that does not work as described above:
```
    it('Triggers a URL change if the hash contains a uPort response', () => {
      global.window = {
        location: { hash: `id=SignRequest&access_token=${res}` },
        onhashchange: sinon.spy(),
      }
      url.getResponse()
      expect(window.onhashchange).to.be.called
    })
```js

Reply:
I don't think it's necessary to go that deep in the testing, since it's basically browser functionality. You can test that no assignment is made though.